sycl/source/detail/global_handler.cpp

```diff
@@ -67,38 +67,39 @@ GlobalHandler::GlobalHandler() = default;
 GlobalHandler::~GlobalHandler() = default;
 
 void GlobalHandler::InitXPTIStuff() {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
   // Let subscribers know a new stream is being initialized
-  getXPTIRegistry().initializeStream(SYCL_SYCLCALL_STREAM_NAME, GMajVer,
-                                     GMinVer, GVerStr);
+  getXPTIRegistry().initializeStream(SYCL_API_STREAM_NAME, GMajVer, GMinVer,
+                                     GVerStr);
   xpti::payload_t SYCLPayload("SYCL Interface Layer");
   uint64_t SYCLInstanceNo;
   GSYCLCallEvent =
       xptiMakeEvent("SYCL API Layer", &SYCLPayload, xpti::trace_algorithm_event,
                     xpti_at::active, &SYCLInstanceNo);
+#endif
 }
 
-bool hasValue(const char *Value) { return Value && Value[0] != '\0'; }
-
 void GlobalHandler::TraceEventXPTI(const char *Message) {
   if (!Message)
     return;
 #ifdef XPTI_ENABLE_INSTRUMENTATION
   if (xptiTraceEnabled()) {
     uint64_t Uid = xptiGetUniqueId(); // Gets the UID from TLS
-    uint8_t StreamID = xptiRegisterStream(SYCL_SYCLCALL_STREAM_NAME);
+    uint8_t StreamID = xptiRegisterStream(SYCL_API_STREAM_NAME);
 
     detail::tls_code_loc_t Tls;
     auto CodeLocation = Tls.query();
     xpti::payload_t Payload(CodeLocation.functionName(),
                             CodeLocation.fileName(), CodeLocation.lineNumber(),
                             CodeLocation.columnNumber(), nullptr);
 
-    uint64_t EventInstanceNo;
+    uint64_t EventInstanceNo{};
     auto Event =
         xptiMakeEvent("SYCL RT exception", &Payload, xpti::trace_diagnostics,
                       xpti_at::active, &EventInstanceNo);
-    xptiNotifySubscribers(StreamID, xpti::trace_diagnostics, GSYCLCallEvent,
-                          Event, Uid, static_cast<const void *>(Message));
+    xptiNotifySubscribers(StreamID, xpti::trace_diagnostics,
+                          static_cast<xpti_td *>(GSYCLCallEvent), Event, Uid,
+                          static_cast<const void *>(Message));
```

Comment:
// We have to handle the cases where: (1) we may have just the code location set 
// and not UID and (2) UID set
detail::tls_code_loc_t Tls;
auto CodeLocation= Tls.query();

// Creating a tracepoint will convert a CodeLocation to UID, if not set
xpti::framework::tracepoint_t TP(CodeLocation.fileName(), CodeLocation.functionName(), CodeLocation.lineNumber(), CodeLocation.ColumnNumber(), nullptr);

// The call to notify will have the signature of:
//   (1) the stream defined in .stream()
//   (2) The trace type equal to what is set by .trace_type()
//   (3) Parent event set to NULL
//   (4) Current event set to one created from CodeLocation and UID
//   (5) An instance ID that records the number of times this code location has been seen
//   (6) The message generated by the exception handler
TP.stream(SYCL_API_STREAM_NAME)
    .trace_type(xpti::trace_diagnostics)
    .event_type(xpti::trace_event_type_t::algorithm)
    .notify(static_cast<const void *>(Message));

Reply:
fixed